pyrenew/process/ar.py

```diff
@@ -52,60 +48,113 @@ def sample(
-            raise ValueError(
-                "Array of autoregressive coefficients "
-                "must be no more than 1 dimension",
-                f"Got {autoreg.ndim}",
+        try:
+            noise_shape = jax.lax.broadcast_shapes(
+                init_vals.shape[1:],
+                autoreg.shape[1:],
+                noise_sd.shape,
             )
-        if not init_vals.ndim == 1:
+        except Exception as e:
             raise ValueError(
-                "Array of initial values must be " "no more than 1 dimension",
-                f"Got {init_vals.ndim}",
-            )
-        order = autoreg.size
-        if not init_vals.size == order:
+                "Could not determine a "
+                "valid shape for the AR process noise "
+                "from the shapes of the init_vals, "
+                "autoreg, and noise_sd arrays. "
+                "See ARProcess documentation for details."
+            ) from e
+
+        if not n_inits == order:
             raise ValueError(
-                "Array of initial values must be "
-                "be the same size as the order of "
-                "the autoregressive process, "
-                "which is determined by the number "
-                "of autoregressive coefficients "
-                "provided. Got {init_vals.size} "
-                "initial values for a process of "
-                f"order {order}"
+                "Initial values array must have the same "
+                "first dimension length as the order p of "
+                "the AR process. The order is given by "
+                "the first dimension length of the array "
+                "of autoregressive coefficients. Got an initial "
+                f"value array with first dimension {n_inits} for "
+                f"a process of order {order}"
             )
 
-        raw_noise = self.noise_rv_(n=n, **kwargs)
-        noise = noise_sd_arr * raw_noise
+        history_shape = (order,) + noise_shape
 
-        def transition(recent_vals, next_noise):  # numpydoc ignore=GL08
-            new_term = jnp.dot(autoreg, recent_vals) + next_noise
-            new_recent_vals = jnp.hstack(
-                [new_term, recent_vals[: (order - 1)]]
-            )
+        try:
+            inits_broadcast = jnp.broadcast_to(init_vals, history_shape)
+        except Exception as e:
+            raise ValueError(
+                "Could not broadcast init_vals "
+                f"(shape {init_vals.shape}) "
+                "to the expected shape of the process "
+                f"history (shape {history_shape}). "
+                "History shape is determined by the "
+                "shapes of the init_vals, autoreg, and "
+                "noise_sd arrays. See ARProcess "
+                "documentation for details"
+            ) from e
+
+        inits_flipped = jnp.flip(inits_broadcast, axis=0)
+        assert jnp.shape(inits_flipped) == history_shape
```

Comment:
Appreciate the informative error messages in this PR, but this one appears to be overlooked.

Reply:
Addressed in 461e024